variation-commons-mongodb/src/test/java/uk/ac/ebi/eva/commons/mongodb/services/VariantWithSamplesAndAnnotationServiceTest.java

```diff
@@ -108,5 +112,33 @@ public void testCountTotalNumberOfVariants() {
         assertEquals(498, service.countTotalNumberOfVariants());
     }
 
+    @Test
+    public void testfindbyRegionAndOtherBeaconFilters() {
+        Region startRange = new Region("9", new Long(10099), new Long(10099));
+        Region endRange = new Region("9", new Long(10099), new Long(10099));
+        List<VariantRepositoryFilter> filters = new FilterBuilder().getBeaconFilters("A", "T", VariantType.SNV, Arrays.asList("PRJEB5829"));
+        List<VariantMongo> variantMongoList = service.findbyRegionAndOtherBeaconFilters(startRange, endRange, filters);
+
+        assertEquals(true, variantMongoList.size() > 0);
+        assertEquals("9", variantMongoList.get(0).getChromosome());
+        assertEquals("A", variantMongoList.get(0).getReference());
+        assertEquals("T", variantMongoList.get(0).getAlternate());
+        assertEquals(VariantType.SNV, variantMongoList.get(0).getType());
+
+
+        filters= new FilterBuilder().getBeaconFilters("A", "T", VariantType.SNV, null);
+        assertEquals(true,service.findbyRegionAndOtherBeaconFilters(startRange, endRange, filters).size()> 0);
+
+        filters= new FilterBuilder().getBeaconFilters("A", "T", null, null);
+        assertEquals(true,service.findbyRegionAndOtherBeaconFilters(startRange, endRange, filters).size()> 0);
+
+        filters= new FilterBuilder().getBeaconFilters("A", null, VariantType.SNV, null);
+        assertEquals(true,service.findbyRegionAndOtherBeaconFilters(startRange, endRange, filters).size()> 0);
+
+        endRange = new Region("9", new Long(10098), new Long(10098));
+        assertEquals(false, service.findbyRegionAndOtherBeaconFilters(startRange, endRange, filters).size() > 0);
+
+    }
```

Comment:
I would suggest adding an extra test using ranges and having different returns when filters are used
```
    @Test
    public void testfindbyRegionAndOtherBeaconFiltersWithRanges() {
        Region startRange = new Region("11", 190238L, 190276L);
        Region endRange = new Region("11", 190238L, 190276L);

        List<VariantRepositoryFilter> filters = new FilterBuilder().getBeaconFilters("A", null, null, null);
        assertEquals(2, service.findbyRegionAndOtherBeaconFilters(startRange, endRange, filters).size());

        filters= new FilterBuilder().getBeaconFilters("A", "T", null, null);
        assertEquals(1, service.findbyRegionAndOtherBeaconFilters(startRange, endRange, filters).size());
        assertEquals("11_190238_A_T", service.findbyRegionAndOtherBeaconFilters(startRange, endRange, filters).get(0).getId());

        filters= new FilterBuilder().getBeaconFilters("A", "C", null, null);
        assertEquals(1, service.findbyRegionAndOtherBeaconFilters(startRange, endRange, filters).size());
        assertEquals("11_190276_A_C", service.findbyRegionAndOtherBeaconFilters(startRange, endRange, filters).get(0).getId());
    }
```

Reply:
I have added this to the test and it is passing as well :)